rust/main/agents/relayer/src/msg/metadata/routing.rs

```diff
@@ -30,10 +31,57 @@ impl MetadataBuilder for RoutingIsmMetadataBuilder {
             .build_routing_ism(ism_address)
             .await
             .map_err(|err| MetadataBuildError::FailedToBuild(err.to_string()))?;
-        let module = ism
-            .route(message)
+
+        let message_domain = KnownHyperlaneDomain::try_from(message.origin)
+            .map(|domain| domain.as_str().to_string())
+            // if its an unknown domain, use the raw u32 as a string
+            .unwrap_or_else(|_| message.origin.to_string());
+        let fn_key = "route";
+
+        // Depending on the cache policy, make use of the message ID
+        let params_cache_key = match self
+            .base_builder()
+            .ism_cache_policy_classifier()
+            .get_cache_policy(self.root_ism, ism.domain(), ModuleType::Routing)
             .await
-            .map_err(|err| MetadataBuildError::FailedToBuild(err.to_string()))?;
+        {
+            // To have the cache key be more succinct, we use the message id
+            IsmCachePolicy::IsmSpecific => (ism.address(), H256::zero()),
```

Comment:
Instead of `H256::zero()` we should have this be the message.origin. The idea is that we want to take advantage of us knowing that it's a DomainRoutingIsm used, which means that we can only cache the result depending on the origin domain of a message. As written this would result in a cache hit for a message from chain A and chain B when we know that we should be getting different results bc they are from a different domain

Might be able to massage the type of `params_cache_key` to tolerate this as a different type without converting to H256 by matching the type expected by `get_cached_call_result`